src/peergos/shared/user/UserContext.java

```diff
@@ -349,17 +350,18 @@ public boolean isJavascript() {
 
     private CompletableFuture<UserContext> init(Consumer<String> progressCallback) {
         progressCallback.accept("Retrieving Friends");
-        return userData.getValue()
-                .thenCompose(wd -> createFileTree(entrie, username, network, crypto.random, crypto.hasher)
+        return writeSynchronizer.getCurrentWriterData(signer.publicKeyHash, signer.publicKeyHash,
+                wd -> createFileTree(entrie, username, network, crypto.random, crypto.hasher)
                         .thenCompose(root -> {
                             this.entrie = root;
                             return getByPath("/" + username + "/" + "shared")
-                                    .thenCompose(sharedOpt -> {
+                                    .thenApply(sharedOpt -> {
```

Comment:
this should still be thenCompose I believe

Reply:
done